warp/backend.go

```diff
@@ -131,100 +121,56 @@ func (b *backend) AddMessage(unsignedMessage *avalancheWarp.UnsignedMessage) err
 		return fmt.Errorf("failed to put warp signature in db: %w", err)
 	}
 
-	var signature [bls.SignatureLen]byte
 	sig, err := b.warpSigner.Sign(unsignedMessage)
 	if err != nil {
 		return fmt.Errorf("failed to sign warp message: %w", err)
 	}
 
-	copy(signature[:], sig)
-	b.messageSignatureCache.Put(messageID, signature)
+	b.signatureCache.Put(messageID, sig)
 	log.Debug("Adding warp message to backend", "messageID", messageID)
 	return nil
 }
 
-func (b *backend) GetMessageSignature(unsignedMessage *avalancheWarp.UnsignedMessage) ([bls.SignatureLen]byte, error) {
+func (b *backend) GetMessageSignature(unsignedMessage *avalancheWarp.UnsignedMessage) ([]byte, error) {
 	messageID := unsignedMessage.ID()
 
 	log.Debug("Getting warp message from backend", "messageID", messageID)
-	if sig, ok := b.messageSignatureCache.Get(messageID); ok {
+	if sig, ok := b.signatureCache.Get(messageID); ok {
 		return sig, nil
 	}
 
-	if err := b.ValidateMessage(unsignedMessage); err != nil {
-		return [bls.SignatureLen]byte{}, fmt.Errorf("failed to validate warp message: %w", err)
+	if err := b.verifyMessage(unsignedMessage); err != nil {
+		return []byte{}, fmt.Errorf("failed to validate warp message: %w", err)
 	}
-
-	var signature [bls.SignatureLen]byte
-	sig, err := b.warpSigner.Sign(unsignedMessage)
-	if err != nil {
-		return [bls.SignatureLen]byte{}, fmt.Errorf("failed to sign warp message: %w", err)
-	}
-
-	copy(signature[:], sig)
-	b.messageSignatureCache.Put(messageID, signature)
-	return signature, nil
+	return b.signMessage(unsignedMessage)
 }
 
-func (b *backend) ValidateMessage(unsignedMessage *avalancheWarp.UnsignedMessage) error {
-	// Known on-chain messages should be signed
-	if _, err := b.GetMessage(unsignedMessage.ID()); err == nil {
-		return nil
-	}
+func (b *backend) GetBlockSignature(blockID ids.ID) ([]byte, error) {
+	log.Debug("Getting block from backend", "blockID", blockID)
```

Comment:
```suggestion
	log.Debug("Getting block signature from backend", "blockID", blockID)
```
also seems this log would be more useful if we emitted some indicator of success/failure